source/slang/slang-ir-metal-legalize.cpp

```diff
@@ -1740,6 +1740,46 @@ namespace Slang
         }
     };
 
+    // metal textures only support writing 4-component values, even if the texture is only 1, 2, or 3-component
```

Comment:
You should be able to use `vectorReshape<>` on the value to write into in stdlib as well? That should remove the need for this leglaization step, right?